cmd/cli/commands/nim.go

```diff
@@ -0,0 +1,353 @@
+	cmd.Printf("Pulling NIM image %s...\n", model)
+
+	reader, err := dockerClient.ImagePull(ctx, model, image.PullOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to pull NIM image: %w", err)
+	}
+	defer reader.Close()
+
+	// Stream pull progress
+	io.Copy(cmd.OutOrStdout(), reader)
+
+	return nil
+}
+
+// findNIMContainer finds an existing NIM container for the given model
+func findNIMContainer(ctx context.Context, dockerClient *client.Client, model string) (string, error) {
+	containerName := nimContainerName(model)
+
+	containers, err := dockerClient.ContainerList(ctx, container.ListOptions{
+		All: true,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to list containers: %w", err)
+	}
+
+	for _, c := range containers {
+		for _, name := range c.Names {
+			if strings.TrimPrefix(name, "/") == containerName {
+				return c.ID, nil
+			}
+		}
+	}
+
+	return "", nil
+}
+
+// createNIMContainer creates and starts a NIM container
+func createNIMContainer(ctx context.Context, dockerClient *client.Client, model string, cmd *cobra.Command) (string, error) {
+	containerName := nimContainerName(model)
+
+	// Get NGC API key from environment
+	ngcAPIKey := os.Getenv("NGC_API_KEY")
+	if ngcAPIKey == "" {
+		cmd.Println("Warning: NGC_API_KEY environment variable is not set. NIM may require authentication.")
+	}
+
+	// Check for GPU support
+	gpu, err := gpupkg.ProbeGPUSupport(ctx, dockerClient)
+	if err != nil {
+		cmd.Printf("Warning: Failed to probe GPU support: %v\n", err)
+		gpu = gpupkg.GPUSupportNone
+	}
+
+	// Create cache directory
+	cacheDir := os.Getenv("LOCAL_NIM_CACHE")
+	if cacheDir == "" {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("failed to get home directory: %w", err)
+		}
+		cacheDir = homeDir + "/.cache/nim"
+	}
+
+	// Create the cache directory if it doesn't exist
+	if err := os.MkdirAll(cacheDir, 0755); err != nil {
+		return "", fmt.Errorf("failed to create NIM cache directory: %w", err)
+	}
+
+	// Container configuration
+	env := []string{}
+	if ngcAPIKey != "" {
+		env = append(env, "NGC_API_KEY="+ngcAPIKey)
+	}
+
+	portStr := strconv.Itoa(nimDefaultPort)
+	config := &container.Config{
+		Image: model,
+		Env:   env,
+		ExposedPorts: nat.PortSet{
+			nat.Port(portStr + "/tcp"): struct{}{},
+		},
+	}
+
+	hostConfig := &container.HostConfig{
+		ShmSize: nimDefaultShmSize,
+		Mounts: []mount.Mount{
+			{
+				Type:   mount.TypeBind,
+				Source: cacheDir,
+				Target: "/opt/nim/.cache",
+			},
+		},
+		PortBindings: nat.PortMap{
+			nat.Port(portStr + "/tcp"): []nat.PortBinding{
+				{
+					HostIP:   "127.0.0.1",
+					HostPort: portStr,
+				},
+			},
+		},
+	}
+
+	// Add GPU support if available
+	if gpu == gpupkg.GPUSupportCUDA {
+		if ok, err := gpupkg.HasNVIDIARuntime(ctx, dockerClient); err == nil && ok {
+			hostConfig.Runtime = "nvidia"
+		}
+		hostConfig.DeviceRequests = []container.DeviceRequest{{
+			Count:        -1,
+			Capabilities: [][]string{{"gpu"}},
+		}}
+	}
```

Comment:
nit: If someone is running NIM, they likely want NVIDIA GPU support. Maybe we should warn the user if we could not detect one?